### PR TITLE
Add pre-commit hook to the repository

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR adds pre-commit hooks to fmt the files before the commit.

You need to install the pre-commit hook by running: 

```bash
make link-git-hooks
                                                                                                                          16.18.0 bazel 5.4.0
==> Installing all git hooks...
find .git/hooks -type l -exec rm {} \;
find .githooks -type f -exec ln -sf ../../{} .git/hooks/ \;
```

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,17 @@
-.PHONY: submit test
-tags=logging callback metrics scheduler
+tags="logging callback metrics scheduler"
 cgo=0 
 goos=linux
 goarch=amd64
-#export TAGS
-#export CGO_ENABLED
-#export GOARCH
```
Comment (colm-quinn) [inline Makefile:8]: 💯

File: pre-commit [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+for FILE in ${STAGED_GO_FILES}
+do
+    gofmt -w -s "${FILE}"
+    goimports -w "${FILE}"
+    git add "${FILE}"
+done
+
+if [[ -n "${STAGED_GO_FILES}" ]]; then
+    set -o errexit
+    go test --tags="unit integration" -race ./internal...
```
Comment (colm-quinn) [inline pre-commit:14]: [q] For my own knowledge, do we have any tests yet?

File: pre-commit [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    make gen-docs > /dev/null
+    git add docs
+fi
+
+STAGED_EVG_FILES=$(git diff --cached --name-only | grep "evergreen.yml$")
+
+for FILE in ${STAGED_EVG_FILES}
+do
+    evergreen validate "${FILE}"
+done
```
Comment (colm-quinn) [inline pre-commit:31]: [nit] Probably a bit early to add this line in, until we have integration with EVG. It's an early optimisation smell.

**Review by colm-quinn — APPROVED** [review]
LGTM

File: pre-commit (hunk 2 above)
Comment (colm-quinn) [reply pre-commit:14]: Actually looking at the codebase, we don't have a `internal` as well as a dual root structure with cfn + cdk folders.

I don't think this line makes sense yet until we add more testing strucutre.

Comment (andreaangiolillo) [reply pre-commit]: sorry I have actually committed the wrong file

Comment (andreaangiolillo) [reply pre-commit]: updated

**Pull request merged**